src/section/shared_state.rs

```diff
@@ -568,6 +578,10 @@ impl SharedState {
                 continue;
             }
 
+            if member_info.is_new_adult() {
+                core.send_event(Event::PromotedToAdult);
```

Comment:
I'd prefer make core sending this notification within the approved.rs, instead of having core passed down.
you could make `increment_age_counters` return a bool indicating whether it `becomes` a new adult